clang/include/clang/CIR/Dialect/IR/CIRAttrs.td

```diff
@@ -202,6 +202,35 @@ def IntAttr : CIR_Attr<"Int", "int", [TypedAttrInterface]> {
   let hasCustomAssemblyFormat = 1;
 }
 
+//===----------------------------------------------------------------------===//
+// FloatAttr
+//===----------------------------------------------------------------------===//
+
+def FloatAttr : CIR_Attr<"Float", "float", [TypedAttrInterface]> {
+  let summary = "An Attribute containing a floating-point value";
+  let description = [{
+    A float attribute is a literal attribute that represents a floating-point
+    value of the specified floating-point type.
+  }];
+  let parameters = (ins AttributeSelfTypeParameter<"">:$type, "APFloat":$value);
+  let builders = [
+    AttrBuilderWithInferredContext<(ins "Type":$type,
+                                        "const APFloat &":$value), [{
+      return $_get(type.getContext(), type, value);
+    }]>,
+    AttrBuilderWithInferredContext<(ins "Type":$type, "double":$value), [{
+      mlir::cir::FloatType fpType = type.cast<mlir::cir::FloatType>();
+      mlir::APFloat apValue(value);
+      return $_get(fpType.getContext(), fpType, apValue);
+    }]>,
+  ];
+  let extraClassDeclaration = [{
+    double getDouble() const { return getValue().convertToDouble(); }
```

Comment:
This method can be later introduced when there's a usecase for it

Reply:
Removed for now.